src/rust/bitbox02-rust-c/src/lib.rs

```diff
@@ -36,14 +36,60 @@ mod sha2;
 mod workflow;
 
 // Whenever execution reaches somewhere it isn't supposed to rust code will "panic". Our panic
-// handler will print the available information on the screen. If we compile with `panic=abort`
-// this code will never get executed.
+// handler will print the available information on the screen and over RTT. If we compile with
+// `panic=abort` this code will never get executed.
 #[cfg(not(test))]
 #[cfg(not(feature = "testing"))]
 #[cfg_attr(feature = "bootloader", allow(unused_variables))]
 #[panic_handler]
 fn panic(info: &core::panic::PanicInfo) -> ! {
+    ::util::log::log!("{}", info);
     #[cfg(feature = "firmware")]
     bitbox02_rust::print_debug!(0, "Error: {}", info);
     loop {}
 }
+
+#[no_mangle]
+pub extern "C" fn rust_rtt_init() {
+    ::util::log::rtt_init();
+}
+
+// SAFETY: The pointer must point to a null terminated string
+#[no_mangle]
+#[cfg_attr(not(feature = "rtt"), allow(unused))]
+pub extern "C" fn rust_log(ptr: *const ::util::c_types::c_char) {
+    #[cfg(feature = "rtt")]
+    {
+        if ptr.is_null() {
+            panic!("`ptr` must be a valid pointer");
+        }
+        let s = unsafe { core::ffi::CStr::from_ptr(ptr as _) };
+        let s = unsafe { core::str::from_utf8_unchecked(s.to_bytes()) };
+        ::util::log::rtt_target::rprintln!("{}", s);
+    }
+}
+
+/// This function is for debug purposes only!
+///
+/// Don't use this in production, the returned pointer is only valid until this function is called
+/// again. This function is not thread safe.
+///
+/// Usage:
+///    uint8_t arr[2] = {1,2};
+///    util_log("%s", rust_hex_encode(rust_util_bytes(arr, sizeof(arr))));
+///
+#[no_mangle]
+#[cfg(feature = "rtt")]
+pub extern "C" fn rust_hex_encode(bytes: util::Bytes) -> *const ::util::c_types::c_char {
```

Comment:
we already have this in util.c

```
void util_uint8_to_hex(const uint8_t* in_bin, const size_t in_len, char* out)
{
    rust_util_uint8_to_hex(
        rust_util_bytes(in_bin, in_len), rust_util_bytes_mut((uint8_t*)out, in_len * 2 + 1));
}
```

with rust_util_uint8_to_hex in util.rs.

Reply:
named it `rust_dbg_hex` and made it even easier to use from C. More in-line with your first comment.